src/editor/Editor.jsx

```diff
@@ -6,49 +6,195 @@ var TextFieldRich = require("../TextFieldRich.jsx");
+                    	<FontIcon onClick={this._onToolbarClick.bind(this, "inserthorizontalrule")} className="cfi cfi-minus" />
+                    	<FontIcon onClick={this._onToolbarClick.bind(this, "link")} className="cfi cfi-link" />
+                    	<FontIcon onClick={this._onToolbarClick.bind(this, "table")} className="cfi cfi-table" />
                     </ToolbarGroup>
+                    <ToolbarGroup key={4} float="left">
+                    	<FontIcon onClick={this._onToolbarClick.bind(this, "insertorderedlist")} className="cfi cfi-list-ol" />
+                    	<FontIcon onClick={this._onToolbarClick.bind(this, "insertunorderedlist")} className="cfi cfi-list-ul" />
+                    	<FontIcon onClick={this._onToolbarClick.bind(this, "forecolor")} className="cfi cfi-eyedropper" />
+                    	<FontIcon onClick={this._onToolbarClick.bind(this, "backcolor")} className="cfi cfi-magic" />
+                    	<div ref="colorPickerContainer" className="chamel-color-picker">
+                    		<div className="chamel-color-picker-close">
+                    			<FontIcon onClick={this._displayColorPicker.bind(this, false, true)} className="cfi cfi-times" />
+                    		</div>
+                    		<div>
+                    			<ColorPicker onChange={this._onColorPick} />
+                    		</div>
+                    	</div>
+                	</ToolbarGroup>
+                    <ToolbarGroup key={5} float="right">
+                        <DropDownIcon 
+                        	iconClassName="cfi cfi-header" 
+                        	menuItems={fontStyleOptions}
+                        	onChange={this._onMenuClick.bind(this, "formatblock")} />
+                        <DropDownIcon 
+	                    	iconClassName="cfi cfi-text-height" 
+	                    	menuItems={fontSizeOptions}
+	                    	onChange={this._onMenuClick.bind(this, "fontsize")} />
+                        <DropDownIcon 
+                        	iconClassName="cfi cfi-font" 
+                        	menuItems={fontNameOptions}
+                    		onChange={this._onMenuClick.bind(this, "fontname")} />
+                </ToolbarGroup>
                 </Toolbar>
                 <div>
-                    <TextFieldRich />
+                    <TextFieldRich ref="rte" />
                 </div>
+                <Dialog ref="linkDialog" title="Enter the link path" actions={dialogActions} modal={true} >
+                	<TextField ref="linkInput" />
+                </Dialog>
             </div>
         )
     },
 
+    /**
+     * Callback used to handle commands when button is clicked on the toolbar
+     *
+     * @param {string} type		The name of the command to execute
+     * @private
+     */
+    _onToolbarClick: function(type) {
+    	if (!this.isMounted()) { 
+    		return;
+    	}
+    	
+    	switch(type) {
+    		case "src":
```

Comment:
This is what I was talking about in chat. It would be awesome if we could have the src and the rte controls be more simple and have the editor just render them here rather than having to use toggleSrc in the rich text component.